packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -468,6 +468,56 @@ class _HelperErrorState extends State<_HelperError> with SingleTickerProviderSta
   }
 }
 
+/// Defines the different IconColor states that the InputDecoration can be in.
+///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// This sample shows how to give an Icon different colors with DecorationIconColors
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return TextFormField(
+///     decoration: const InputDecoration(
+///         icon: Icon(Icons.category),
+///         decorationIconColors: DecorationIconColors(
+///             iconColor: Colors.green,
+///             active: Colors.orange,
+///             disabled: Colors.red,
+///             hover: Colors.amber
+///         )
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [InputDecoration] where it is used
+class DecorationIconColors {
```

Comment:
This looks good. However, we have been trying to transition to using `MaterialStateProperty<Color>` for things like this. It allows us to have a single property instead of multiple properties for each state. It also allows the application to customize the state presentation to what ever level they like. If the app wants to specify a color for "hover and disabled" in addition to just hover or disabled, the MaterialStateProperty would allow them to do this.

Can you take a look at this class and see if it would make sense to use it instead of `DecorationIconColors`? To see an example of how it is being used, take a look at the newish `ButtonStyle` class which makes extensive use of this.

Reply:
No worries regarding the delay. Great idea, thank you. I did not know that this class exists 👍. I will have a look, and I can come up with a new solution.